app/portainer/components/datatables/amt-devices-datatable/amtDevicesDatatableController.js

```diff
@@ -52,16 +53,20 @@ angular.module('portainer.docker').controller('AMTDevicesDatatableController', [
         if (!confirmed) {
           return;
         }
-        this.state.executingAction[deviceGUID] = true;
+        $scope.$evalAsync(() => {
```

Comment:
just wrap the whole function IMO

Reply:
I tried to do it that way but it didn't work, I think because wrapping the whole function makes it wait for `executeDeviceAction` to return before updating the angular state, resulting in inconsistent UI

Reply:
update: it worked using the $async service function instead of $evalAsync

Reply:
@chiptus  I still wasn't able to implement this wrapping the entire `executeDeviceActionConfirm` function with $async, but I at least left a cleaner version with a single use of the async function